lib/iaas/GcpClient.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+const assert = require('assert');
+const _ = require('lodash');
+const Promise = require('bluebird');
+const GcpStorage = require('@google-cloud/storage');
+const GcpCompute = require('@google-cloud/compute');
+const logger = require('../logger');
+const errors = require('../errors');
+const utils = require('../utils');
+const BaseCloudClient = require('./BaseCloudClient');
+const NotFound = errors.NotFound;
+const Unauthorized = errors.Unauthorized;
+const Forbidden = errors.Forbidden;
+
+class GcpClient extends BaseCloudClient {
+  constructor(settings) {
+    super(settings);
+    this.constructor.validateParams(this.settings);
```

Comment:
should it be this.constructor.validateParams(this.settings) --> this.constructor.validateParams(_.chain(this.settings).value()) ??

Reply:
I tested this locally, it works with simple settings param. I have also added unit tests for the same.